python/demo/undocumented/curl-curl/demo_curl-curl.py

```diff
@@ -37,7 +37,7 @@
     print("This demo requires PETSc to be configured with HYPRE.")
     exit()
 
-if not has_petsc4py():
+if not config.has_petsc4py:
```

Comment:
Isn't petsc4py a required dependency too?

Reply:
I don't think so. See `python/CMakeLists.txt`.

Reply:
petsc4py isn't a required dependency. I would be reluctant to make it requires because it's not always in sync with PETSc master.

Reply:
I asked this because [`python/doc/source/installation.rst`](https://github.com/FEniCS/dolfinx/blob/896e36c46ac12d91b696d8f336dbddb17fc84b29/python/doc/source/installation.rst) states that petsc4py is a required dependency for the python interface.

Reply:
@jhale might have added that to the docs.